### DISCUSSION TITLE
App crashed if result of JavascriptObject is to large. What can I do?

### DISCUSSION THREAD

**Discussion by javadevil1982** [cefsharp/CefSharp#4819] [Q&A]
I use CefSharp (CefSharp.WinForms 123.0.60) in a dotnet 4.8 Framework WinForms application. For resons it must be x86. 

I run the CefBrowser with:
JavascriptObjectRepository.Settings.LegacyBindingEnabled = true;
JavascriptObjectRepository.Register("MyClass", MyClassInstance, true, BindingOptions.DefaultBinder);

If I now address MyClass in Javascript and call a method whose result is apparently too large, the whole application crashes. So up to the end of the dotnet method everything works, after the return of the result the crash occurs. Even though I will try to avoid such a large result in the future, I would still like to know if there is a way to catch and handle such errors in the CefSharp CefBrowser instead of crashing the whole application?

**Comment by amaitland** [comment]
>  For resons it must be x86.

You can always run the browser in a separate process so you can use a 64bit browser. There's an example at https://github.com/cefsharp/CefSharp.OutOfProcess

There are other browsers that run out of process by default that might better suite your purposes.

>  handle such errors in the CefSharp CefBrowser instead of crashing the whole application?

Without knowing exactly where it's crashing it's impossible to say. It might be possible to add some additional error handling. You are welcome to submit a PR with proposed improvements.

**Reply by amaitland** [reply]
Also make sure your app is large address aware.

https://github.com/cefsharp/CefSharp/wiki/General-Usage#win32-out-of-memory

**Reply by javadevil1982** [reply]
Thanks for the answers. They sound very promising, but unfortunately I am currently very busy with another project. I will write about my experiences with the options mentioned as soon as I get the chance to try them out.
